org.scala-ide.sdt.core.tests/src/org/scalaide/core/sbtbuilder/ScalaJavaDepTicket_1000607Test.scala

```diff
@@ -0,0 +1,60 @@
+package org.scalaide.core
+package sbtbuilder
+
+import org.junit.Before
+import org.junit.Assert._
+import org.scalaide.core.testsetup.SDTTestUtils
+import org.junit.Test
+import org.eclipse.core.runtime.NullProgressMonitor
+import org.eclipse.core.resources.IncrementalProjectBuilder
+import org.eclipse.core.resources.IFile
+
+object ScalaJavaDepTicket_1000607Test extends testsetup.TestProjectSetup("scalajavadepticket1000607")
+
+/**
+ * Check the test case from the assembla ticket 1000607.
+ * Start from a setup with a compilation error, 'fix' it in the Scala file and check that
+ * the incremental compilation correctly recompile the Java file.
+ */
+class ScalaJavaDepTicket_1000607Test {
+
+  import ScalaJavaDepTicket_1000607Test._
+
+  @Before
+  def setupWorkspace(): Unit = {
+    SDTTestUtils.enableAutoBuild(false)
+  }
+
+  @Test def ticket_1000607(): Unit = {
+    val aClass = compilationUnit("ticket_1000607/A.scala")
+    val cClass = compilationUnit("ticket_1000607/C.java")
+
+    def getProblemMarkers = SDTTestUtils.getProblemMarkers(aClass, cClass)
+
+    // do a clean build and check the expected error
+    cleanProject
+    var problems = getProblemMarkers
+    assertEquals("One error expected: " + SDTTestUtils.userFriendlyMarkers(problems), 1, problems.size)
```

Comment:
Please make it consistent with other assertions. Message itself should make sense. I.e. "One build error expected, got: <list of errors>"